src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileProperty.java

```diff
@@ -6,31 +6,49 @@
 import hudson.Extension;
 import hudson.model.*;
 import net.sf.json.JSONObject;
+import org.jenkinsci.Symbol;
 import org.jenkinsci.lib.configprovider.ConfigProvider;
 import org.jenkinsci.lib.configprovider.model.Config;
 import org.jenkinsci.plugins.configfiles.ConfigByIdComparator;
 import org.jenkinsci.plugins.configfiles.ConfigByNameComparator;
 import org.jenkinsci.plugins.configfiles.ConfigFileStore;
 import org.jenkinsci.plugins.configfiles.ConfigProviderComparator;
+import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.StaplerRequest;
 
 import java.io.IOException;
 import java.util.*;
+import java.util.stream.Collectors;
 
 public class FolderConfigFileProperty extends AbstractFolderProperty<AbstractFolder<?>> implements ConfigFileStore {
 
     private static Comparator<Config> COMPARATOR = new ConfigByIdComparator();
 
     private static ConfigProviderComparator CONFIGPROVIDER_COMPARATOR = new ConfigProviderComparator();
 
-    private Collection<Config> configs = new TreeSet<>(COMPARATOR);
+    private Collection<Config> configs;
 
-    private transient AbstractFolder<?> owner;
+    /*package*/ FolderConfigFileProperty() {
+        this((Collection<Config>) null);
+    }
 
+    @Deprecated
     /*package*/ FolderConfigFileProperty(AbstractFolder<?> owner) {
```

Comment:
given this is package-private scope - should all references not be be updated and this removed rather then leaving deprecated as they are only in this code.